includes/class-smaily-for-wp.php

```diff
@@ -0,0 +1,225 @@
+	 * - Smaily_For_WP_Widget.   Defines the widget functionality.
+	 * - Smaily_For_WP_Public.   Defines all hooks for the public side of the site.
+	 *
+	 * Create an instance of the loader which will be used to register the hooks
+	 * with WordPress.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 */
+	private function load_dependencies() {
+		require_once SMLY4WP_PLUGIN_PATH . '/admin/class-smaily-for-wp-admin.php';
+		require_once SMLY4WP_PLUGIN_PATH . '/includes/class-smaily-for-wp-i18n.php';
+		require_once SMLY4WP_PLUGIN_PATH . '/includes/class-smaily-for-wp-loader.php';
+		require_once SMLY4WP_PLUGIN_PATH . '/includes/class-smaily-for-wp-request.php';
+		require_once SMLY4WP_PLUGIN_PATH . '/includes/class-smaily-for-wp-template.php';
+		require_once SMLY4WP_PLUGIN_PATH . '/includes/class-smaily-for-wp-widget.php';
+		require_once SMLY4WP_PLUGIN_PATH . '/public/class-smaily-for-wp-public.php';
+		$this->loader = new Smaily_For_WP_Loader();
+	}
+
+	/**
+	 * Define the locale for this plugin for internationalization.
+	 *
+	 * Uses the Smaily_For_WP_i18n class in order to set the domain and to register the hook
+	 * with WordPress.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 */
+	private function set_locale() {
+		$plugin_i18n = new Smaily_For_WP_i18n();
+
+		$this->loader->add_action( 'plugins_loaded', $plugin_i18n, 'load_plugin_textdomain' );
+	}
+
+	/**
+	 * Register all of the hooks related to the admin area functionality
+	 * of the plugin.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 */
+	private function define_admin_hooks() {
+		$plugin_admin = new Smaily_For_WP_Admin( $this->get_plugin_name(), $this->get_version() );
+		$this->loader->add_action( 'admin_enqueue_scripts', $plugin_admin, 'enqueue_styles' );
+		$this->loader->add_action( 'admin_enqueue_scripts', $plugin_admin, 'enqueue_scripts' );
+		$this->loader->add_action( 'wp_ajax_smaily_admin_save', $plugin_admin, 'smaily_admin_save' );
+		$this->loader->add_action( 'widgets_init', $plugin_admin, 'smaily_subscription_widget_init' );
+		$this->loader->add_action( 'admin_menu', $plugin_admin, 'smaily_admin_render' );
+	}
+
+	/**
+	 * Register all of the hooks related to the public-facing functionality
+	 * of the plugin.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 */
+	private function define_public_hooks() {
+		$plugin_public = new Smaily_For_WP_Public( $this->get_plugin_name(), $this->get_version() );
+		$this->loader->add_action( 'init', $plugin_public, 'add_shortcodes' );
+	}
+
+	/**
+	 * Install database structure (on activation).
+	 *
+	 * @since 3.0.0
+	 */
+	public static function activate() {
+		global $wpdb;
+		require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
+		$charset_collate = $wpdb->get_charset_collate();
+
+		// Create database table - settings.
+		$table_name            = esc_sql( $wpdb->prefix . 'smaily_config' );
+		$settings_table_exists = $wpdb->get_var(
+			$wpdb->prepare(
+				'SHOW TABLES LIKE %s',
+				$table_name
+			)
+		);
```

Comment:
This is no longer used and can be removed.

Reply:
96b354a